packages/syft/src/syft/ast/klass.py

```diff
@@ -617,10 +619,41 @@ def create_pointer_class(self) -> None:
         attrs["__name__"] = name
         attrs["__module__"] = ".".join(parts)
 
-        klass_pointer = type(self.pointer_name, (Pointer,), attrs)
+        # if the object already has a pointer class specified, use that instead of creating
+        # an empty subclass of Pointer
+        if hasattr(self.object_ref, "PointerClassOverride"):
+
+            klass_pointer = getattr(self.object_ref, "PointerClassOverride")
+            for key, val in attrs.items():
+
+                # only override functioanlity of AST attributes if they
+                # don't already exist on the PointerClassOverride class
+                # (the opposite of inheritance)
+                if not hasattr(klass_pointer, key):
+                    setattr(klass_pointer, key, val)
+                else:
+                    # TODO: cache attribute in backup_ location so that we can use them if we want
```

Comment:
Add the backup attr cache to the backlog? This relates to calling super class op implementations like `__add__` when implementing subclass Tensor types.